doc/gas-handling.md

```diff
@@ -0,0 +1,53 @@
+# Gas Handling And Weight Costs In Pallet-Move
+
+The gas unit is an internal processing fee for publishing modules and executing scripts within the MoveVM.
+
+In Substrate-based blockchains, another unit, `Weight`, measures and manages the time it takes to validate a block.
+Substrate defines one unit of weight as one picosecond of execution time on reference hardware.
+
+## Gas handling in Move layer
+
+From within the MoveVM, there are two different sources for gas costs:
+- Cost for script execution,
+- Cost for publishing modules and bundles.
+
+Script execution can broken down into:
+- Gas cost per bytecode instruction - bytecode gas cost table can be found [here](https://github.com/eigerco/substrate-move/blob/d33762e0ec9ee82ca71d7c5991247eec135666d1/move-vm-backend-common/src/gas_schedule.rs#L39)
+- Gas cost per native function call with variable gas cost depending on the input function arguments - native gas cost table can be found [here](https://github.com/eigerco/substrate-move/blob/d33762e0ec9ee82ca71d7c5991247eec135666d1/move-vm-backend-common/src/gas_schedule.rs#L166)
+
+It is possible to modify the table values to control the cost of script execution.
+It is recommended that both tables be scaled by the same factor to preserve the proportional cost relationship between native functions and bytecode instructions.
+These table values are less dependent on storage access and are more related to raw processing time.
+
+The cost for publishing modules and bundles is independently defined from the script execution costs and it can be controlled by modifying [gas-cost-per-byte constant](https://github.com/eigerco/substrate-move/blob/d33762e0ec9ee82ca71d7c5991247eec135666d1/move-vm-backend-common/src/gas_schedule.rs#L23).
+This value was selected during the testing and benchmarks and it is an arbitrary choice specifically for the MoveVM instance running within the Substrate runtime.
+
+All internal MoveVM gas handling costs are defined in the same [gas schedule module](https://github.com/eigerco/substrate-move/blob/main/move-vm-backend-common/src/gas_schedule.rs) so that tweaking any gas-related factor can be done from within that module.
+
+## Extrinsic Weight Cost in Pallet layer
+
+Three main extrinsics interact with MoveVM, which stores its state within the Substrate storage:
+- `publish_module`
+- `publish_bundle`
```

Comment:
Wasn't it `publish_module_bundle` or did you rename it?

Reply:
Oh right, good catch!